src/AspNet.Security.OAuth.Ebay/EbayAuthenticationHandler.cs

```diff
@@ -0,0 +1,146 @@
+
+            return QueryHelpers.AddQueryString(this.Options.AuthorizationEndpoint, parameters!);
+        }
+
+        protected override async Task<OAuthTokenResponse> ExchangeCodeAsync([NotNull] OAuthCodeExchangeContext context)
+        {
+            using var request = new HttpRequestMessage(HttpMethod.Post, this.Options.TokenEndpoint);
+            request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            request.Headers.Authorization = CreateAuthorizationHeader();
+
+            var parameters = new Dictionary<string, string>
+            {
+                ["grant_type"] = "authorization_code",
+                ["code"] = context.Code,
+                ["redirect_uri"] = this.Options.RuName!
+            };
+
+            request.Content = new FormUrlEncodedContent(parameters!);
+
+            using var response = await this.Backchannel.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, this.Context.RequestAborted);
+            if (!response.IsSuccessStatusCode)
+            {
+                Logger.LogError("An error occurred while retrieving an access token: the remote server " +
+                                "returned a {Status} response with the following payload: {Headers} {Body}.",
+                                /* Status: */ response.StatusCode,
+                                /* Headers: */ response.Headers.ToString(),
+                                /* Body: */ await response.Content.ReadAsStringAsync(this.Context.RequestAborted));
+
+                return OAuthTokenResponse.Failed(new Exception("An error occurred while retrieving an access token."));
+            }
+
+            var payload = JsonDocument.Parse(await response.Content.ReadAsStringAsync(this.Context.RequestAborted));
+
+            return OAuthTokenResponse.Success(payload);
+        }
+
+        private AuthenticationHeaderValue CreateAuthorizationHeader()
+        {
+            string credentials = Convert.ToBase64String(Encoding.ASCII.GetBytes(
+                string.Concat(
+                    EscapeDataString(this.Options.ClientId),
+                    ":",
+                    EscapeDataString(this.Options.ClientSecret))));
+
+            return new AuthenticationHeaderValue("Basic", credentials);
+        }
+
+        private static string EscapeDataString(string value)
+        {
+            if (string.IsNullOrEmpty(value))
+            {
+                return string.Empty;
+            }
+
+            return Uri.EscapeDataString(value).Replace("%20", "+", StringComparison.Ordinal);
+        }
+
+        private static string NormalizeScope(string scope)
+        {
+            if (!scope.StartsWith(EbayAuthenticationDefaults.ScopePrefix, StringComparison.OrdinalIgnoreCase))
+            {
+                scope = string.Format(CultureInfo.InvariantCulture, "{0}/{1}", EbayAuthenticationDefaults.ScopePrefix, scope);
+            }
+
+            return scope;
```

Comment:
Why is this even needed? The scopes should just be declared with their correct names, rather than patched on-the-fly to have the correct prefix.

Reply:
Ebay supports a sandbox env, between prod env and sandbox env there is a different host. This prompted me to write this method. Checking the documentation again I saw that the scopes remain the same between the two environments so both this and [discussion_r716010791](https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/pull/603#discussion_r716010791), [discussion_r716009765]9https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/pull/603#discussion_r716009765) can be resolve as you suggested.